data_collection/web/guides.rst

```diff
@@ -0,0 +1,717 @@
+
+
+Examples of download link URL:
+
+ - file extension is at the very end of path (eg. ``http://example.com/file.7z`` or ``http://example.com/article/file.7z?source=user#how-to``)
+ - file extension is at the end of query param value (eg. ``http://example.com/article?click=file.7z&page=3`` or ``http://example.com/article?target=file.7z#how-to``)
+
+Customizing list of file extensions
+"""""""""""""""""""""""""""""""""""
+
+You can customize list of file extensions you want to track as downloads. For example, if you want to track only images as downloads, you can use :ref:`setDownloadExtension<jtc-api-setDownloadExtension>` function to replace the list like this:
+
+.. code-block:: javascript
+
+  // track clicks on images links (eg. <a href="image.png">) only
+  _paq.push(["setDownloadExtensions", "png|jpg|webp|gif"]);
+
+You can add new extensions, to an existing list with ``addDownloadExtensions``:
+
+.. code-block:: javascript
+
+  // add other image formats
+  _paq.push(["addDownloadExtensions", "svg|xcf"]);
+
+Or remove some of extensions from the existing list with ``removeDownloadExtensions``:
+
+.. code-block:: javascript
+
+  _paq.push(["removeDownloadExtensions", "jpg|jpeg"]);
+
+
+Manually marking links as downloads
+"""""""""""""""""""""""""""""""""""
+
+If your download link can't be detected by extension, you still can tell tracker that link should be tracked as a download.
+
+You can add a ``download`` attribute to a link HTML tag. eg.
+
+.. code-block:: html
+
+  <a href="/target-file" download>
+
+Or if you have to be strict with your HTML, you can add a HTML tag class. Default classes are ``piwik_download`` and ``piwik-download``. Eg.
+
+.. code-block:: html
+
+  <a href="/taget-file" class="piwik-download">
+
+Additionally you can define your custom CSS classes for download links with our :ref:`JavaScript Tracking Client API<jtc-api-setDownloadClasses>`. Eg.
+
+.. code-block:: javascript
+
+  _paq.push(["setDownloadClasses", "custom-download-class"]);
+  _paq.push(["trackPageView"]);
+
+or you can define a list of classes at once, by passing an array list of CSS classes:
+
+.. code-block:: javascript
+
+  _paq.push(["setDownloadClasses", ["custom-download-class", "other-download-class", "another-class"]]);
+  _paq.push(["trackPageView"]);
+
+and use that class in HTML code:
+
+.. code-block:: html
+
+  <a href="/taget-file" class="custom-download-class">
+
+.. note::
+  You have to remember that using ``setDownloadClasses`` always overwrite current list of CSS classes.
+
+
+Tracking downloads with inline Javascript
+"""""""""""""""""""""""""""""""""""""""""
+
+There is another alternative for above methods. You can track a download with inline JavaScript code. Insert inline code to HTML tag with ``onclick`` attribute:
+
+.. code-block:: html
+
+  <a href="https://piwik.pro/document-url" target="_blank" onClick="_paq.push(['trackLink', 'https://piwik.pro/document-url', 'download']);">Download document</a>
+
+.. hide::
```

Comment:
This directive doesn't exist. What was intention of adding it?

Reply:
It may be added by some editor. Whatever the reason was, let's just remove it. I will push the change.

Reply:
Done